doc/api/child_process.md

```diff
@@ -1061,6 +1061,9 @@ pipes between the parent and child. The value is one of the following:
    corresponds to the index in the `stdio` array. The stream must have an
    underlying descriptor (file streams do not start until the `'open'` event has
    occurred).
+   **NOTE:** You cannot pass `stdin` as a writable or `stdout`/`stderr` as readable
+   under `options.stdio`. If you do so on a stream object, there is no guarantee
+   that the callback will be called. If the stream errors, all callbacks will be dropped.
```

Comment:
IIUC you *can* pass a readable where a writable should be (and vise-versa), however it's not recommended, right?

Reply:
@RedYetiDev  It seems from further research, you can technically pass a readable stream where a writable is expected (and vice-versa) in Node.js, but it's not recommended because:

1. **Stream Behavior**: Readable and writable streams have distinct behaviors. A readable stream emits data that can be consumed, whereas a writable stream receives data to process. Swapping them might cause unexpected results, errors, or simply do nothing if the stream isn't used correctly.

2. **Compatibility Issues**: Some operations depend on specific stream types. For example, trying to write to a readable stream or read from a writable stream could result in runtime errors or program malfunction since the intended stream behavior is different.

3. **Design Consistency**: Keeping streams properly defined ensures that your code is easier to read, debug, and maintain. It's best to follow the convention that stdin is writable and stdout/stderr are readable, which aligns with how processes and streams are designed to work in Node.js.

# Example of potential issues:

- If you mistakenly treat stdout (which should be readable) as writable, you'd get an error when trying to write to it.

- Similarly, if you pass a readable stream (like stdout) to something expecting writable input (like stdin), it would do nothing since stdout cannot accept writes.

Overall doing this is strongly discouraged and should always use streams as intended ((readable for output, writable for input) to avoid unexpected behaviors and erros. 

I can edit the docs to reflects this discovery.